gateway_code/open_nodes/node_rtl_sdr.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding:utf-8 -*-
+
+# This file is a part of IoT-LAB gateway_code
+# Copyright (C) 2015 INRIA (Contact: admin@iot-lab.info)
+# Contributor(s) : see AUTHORS file
+#
+# This software is governed by the CeCILL license under French law
+# and abiding by the rules of distribution of free software.  You can  use,
+# modify and/ or redistribute the software under the terms of the CeCILL
+# license as circulated by CEA, CNRS and INRIA at the following URL
+# http://www.cecill.info.
+#
+# As a counterpart to the access to the source code and  rights to copy,
+# modify and redistribute granted by the license, users are provided only
+# with a limited warranty  and the software's author,  the holder of the
+# economic rights,  and the successive licensors  have only  limited
+# liability.
+#
+# The fact that you are presently reading this means that you have had
+# knowledge of the CeCILL license and that you accept its terms.
+
+""" Open Node Rtl SDR experiment implementation """
+
+import logging
+import shlex
+
+from gateway_code.common import logger_call
+from gateway_code.utils import subprocess_timeout
+from gateway_code.utils.rtl_tcp import RtlTcp
+from gateway_code.open_nodes.common.node_no import NodeNoBase
+
+LOGGER = logging.getLogger('gateway_code')
+
+# This command controls the power of the open node/rtl_tcp USB stick via the
+# Yepkit module.
+YKUSHCMD = "sudo ykushcmd {model} {cmd} {port}"
+
+
+class NodeRtlSdr(NodeNoBase):
+    """ Open node Rtl SDR implementation """
+
+    TYPE = 'rtl_sdr'
+    RTL_TCP_PORT = 50000
+    RTL_TCP_FREQ = 868000000
```

Comment:
hardcoded to 868 MHz I see, wouldn't it make more sense to have this be configurable at experiment submit by the user ? Or maybe we can send something on port 50000 to change it once the experiment is started ?

Reply:
This default value is selected for convenience but in the end, it doesn't make any difference: in GNU Radio, the configuration box change it to what the user want. I expect people to use a tool like GNU Radio.